src/modules/mailer/domain/factories/email.factory.ts

```diff
@@ -0,0 +1,32 @@
+import { Email } from '../entities/email.entity';
+
+export class EmailFactory {
+  static create(data: {
+    to: string[];
+    subject?: string;
+    text?: string;
+    html?: string;
+  }) {
+    const { to, subject, text, html } = data;
+
+    this.validateEmailAddresses(to);
+
+    return new Email(to, subject, text, html);
```

Comment:
Non-blocking: And here is the continuation of the previous comment: probably, that would be nice to wrap these parameters into an object as well. Thus, no one would need to know/remember which order the parameters should be passed in, and no destcruturing step would be needed)